src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnicodeRanges.cs

```diff
@@ -32,9 +32,8 @@ private static UnicodeRange CreateEmptyRange(ref UnicodeRange range)
         {
             // If the range hasn't been created, create it now.
             // It's ok if two threads race and one overwrites the other's 'range' value.
-            var newRange = new UnicodeRange(0, 0);
-            Volatile.Write(ref range, newRange);
-            return newRange;
+            var range = new UnicodeRange(0, 0);
+            return range;
```

Comment:
I don't think we need it. CLR must ensure partially instantiated objects are not visible outside of their ctors, and the assignment is atomic as it's a reference type.